esphome/components/lvgl/__init__.py

```
@@ -0,0 +1,229 @@
+)
+
+WIDGET_SCHEMA = any_widget_schema()
+
+
+async def add_init_lambda(lv_component, init):
+    if init:
+        lamb = await cg.process_lambda(Lambda(init), [(lv_disp_t_ptr, "lv_disp")])
+        cg.add(lv_component.add_init_lambda(lamb))
+
+
+lv_defines = {}  # Dict of #defines to provide as build flags
+
+
+def add_define(macro, value="1"):
+    if macro in lv_defines and lv_defines[macro] != value:
+        LOGGER.error(
+            "Redefinition of %s - was %s now %s", macro, lv_defines[macro], value
+        )
+    lv_defines[macro] = value
+
+
+def as_macro(macro, value):
+    if value is None:
+        return f"#define {macro}"
+    return f"#define {macro} {value}"
+
+
+LV_CONF_FILENAME = "lv_conf.h"
+LV_CONF_H_FORMAT = """\
+#pragma once
+{}
+"""
+
+
+def generate_lv_conf_h():
+    definitions = [as_macro(m, v) for m, v in lv_defines.items()]
+    definitions.sort()
+    return LV_CONF_H_FORMAT.format("\n".join(definitions))
+
+
+def get_display_list(config):
+    result = []
+    display_list = config.get(df.CONF_DISPLAYS)
+    if isinstance(display_list, list):
+        for display in config.get(df.CONF_DISPLAYS, []):
+            result.append(display[CONF_DISPLAY_ID])
+    else:
+        result.append(display_list)
+    return result
+
+
+def warning_checks(config):
+    global_config = CORE.config
+    displays = get_display_list(config)
+    if display_conf := global_config.get(CONF_DISPLAY):
+        for display_id in displays:
+            display = list(
+                filter(lambda c, k=display_id: c[CONF_ID] == k, display_conf)
+            )[0]
+            if CONF_LAMBDA in display:
+                LOGGER.warning(
+                    "Using lambda: in display config not recommended with LVGL"
+                )
+            if display[CONF_AUTO_CLEAR_ENABLED]:
+                LOGGER.warning(
+                    "Using auto_clear_enabled: true in display config not recommended with LVGL"
+                )
```

Is there a case to allow this to be set to true? If not, I would raise an Invalid error. Same with the lambda above.